### PR TITLE
Part 1: adjust to lark

### PR DESCRIPTION
**Pull request by gimre-xymcity**

### CODE REVIEW

File: generator/main.py [hunk 1]
```diff
@@ -91,39 +91,25 @@ def create_printer(type_instance, name):
...
-            return type_instance
+def process_struct(type_map, struct_type: StructObject, abstract_impl_map: AbstractImplMap):
+    if 'factory_type' in struct_type.yaml_descriptor:
+        factory_type = type_map[struct_type.yaml_descriptor['factory_type']]
+        fix_item(factory_type.yaml_descriptor, 'discriminator')
 
-    return None
+        initializers = factory_type.yaml_descriptor.get('initializers', None)
+        for init in initializers:
+            fix_item(init, 'target_property_name')
```
Comment (gimre-xymcity) [inline generator/main.py:101]: not super pretty, bit unsure when/where this should happen

File: generator/type_objects.py [hunk 2]
```diff
@@ -67,9 +67,9 @@ class StructObject(BaseObject):
     def __init__(self, yaml_descritor):
         super().__init__('struct', yaml_descritor)
         self.layout = []
-        self.has_inlines = False
+        self.dynamic_size = self.yaml_descriptor.get('size', None)
```
Comment (gimre-xymcity) [inline generator/type_objects.py:70]: yay 1!

File: generator/type_objects.py [hunk 3]
```diff
@@ -67,9 +67,9 @@ class StructObject(BaseObject):
...
         self.name_to_index = {}
-        self.is_abstract = self.typename in ['Transaction', 'EmbeddedTransaction']
+        self.is_abstract = 'abstract' == self.yaml_descriptor.get('disposition', None)
```
Comment (gimre-xymcity) [inline generator/type_objects.py:72]: yay 2!

**Review by Jaguar0625 — APPROVED** [review]

File: generator/main.py (hunk 1 above)
Comment (Jaguar0625) [reply generator/main.py:101]: inside AbstractImplMap ?

**Comment by Jaguar0625** [comment]
travis is failing (not sure if expected)

**Review by Jaguar0625 — APPROVED** [review]

**Comment by gimre-xymcity** [comment]
merged into 20b2ea190e571ac79bf3d9fb6bb3af5359422095

**Pull request closed without merging**